PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -815,6 +836,94 @@ double PATMuonProducer::getRelMiniIsoPUCorrected(const pat::Muon& muon, float rh
   return pat::muonRelMiniIsoPUCorrected(muon.miniPFIsolation(), muon.p4(), drcut, rho);
 }
 
+
+double PATMuonProducer::puppiCombinedIsolation(const pat::Muon& muon, const pat::PackedCandidateCollection *pc)
+{
+  double dR_threshold = 0.4;
+  double dR2_threshold = dR_threshold * dR_threshold ;
+  double mix_fraction = 0.5;
+  const std::vector<pat::PackedCandidate> * candidates = pc;
+  enum particleType{
+      CH = 0,
+      NH = 1,
+      PH = 2,
+      OTHER = 100000
+    };
+  double val_PuppiWithLep    [3]= {0,0,0} ;
+  double val_PuppiWithoutLep [3]= {0,0,0} ;  
```

Comment:
these arrays appear not necessary: all three values are simply added in the end. A scalar value is enough.

Reply:
@slava77  done! #26233